### ISSUE TITLE
kms-windows port interest group

### ISSUE THREAD

**Issue by sledz** [kirushyk/kms-windows#14]
Unfortunately this project has fallen asleep since @kirushyk has finished the work here more or less.

On the other side there are some people interested in this port. Our company is one of them.

It would be very good to update the port to the latest Kurento version which has fixed a lot of bugs, a lot of features are still missing (e.g. H.264, see #4), the build process is a little horror, and so on.

At the moment our company does not have the manpower to work on these tasks ourselfs. So we are looking for other ones who are willing to form some kind of an interest group to finance some external ressources doing the work.

Interested parties please reply here or contact kms-windows@dresearch-fe.de!

**Comment by kirushyk** [comment]
Oh, that sounds terrible. I can assist, but I can not work on this. I just have other priorities.

**Comment by sledz** [comment]
@luismcosta, @chris64digital, @petergerten, @Nintha, @Stefan-Z-Camilleri, @devblissit, @ArJavBi, I just want to make you aware of this issue here.

**Comment by stefan-camilleri** [comment]
Hi @sledz , this is a good initiative.  Unfortunately I do not have that much bandwidth right now, so not sure how much time I can dedicate to this.

That said, I am willing to help as I can.

**Comment by kirushyk** [comment]
I think best way to proceed with this is to contact Kurento team with me so we can merge this into their project.

**Comment by pschuessler** [comment]
Ciao @kirushyk, you write "...to contact Kurento team with me so we can merge..."
Does it mean you would/could support us contacting kurento team for further proceeding?
Greetings, Peter (from DResearch team)

**Comment by kirushyk** [comment]
No,

I just meant that's good idea in my opinion: to merge to main project.

**Comment by pschuessler** [comment]
Okay, misunderstanding. Do you know any commercial parties/companies who are interested in pushing forward this issue?

**Comment by kirushyk** [comment]
Looks like one who created this issue is interested :-) I am not sure I know somebody else. Maybe company named ESTOS.

**Comment by pschuessler** [comment]
Thats true ;o)
Do you know anybody I could contact at ESTOS?

**Comment by petergerten** [comment]
I would be happy with the current state - if a built would work. Did anyone get that far? (also see https://github.com/kirushyk/kms-windows/issues/13)

**Comment by pschuessler** [comment]
@kirushyk: could you give a rough estimation of development effort expected for finishing the project (days/weeks)?

**Comment by kirushyk** [comment]
What do you mean?

We can declare it finished right now.

**Comment by petergerten** [comment]
@kirushyk I would agree - but it seems that nobody can solve the issue / bug here: https://github.com/kirushyk/kms-windows/issues/13 
And you seem to be too busy unfortunately. It would be great if you have a pointer in the right direction :)

**Comment by pschuessler** [comment]
@kirushyk: okay, understood. Finsihed (except bug #13) but without additional feature we need for our purposes... Could you imagine to join a type of "extention project" in role as assistant (with limited effort on your side due to your current situation) for support to ensure the developers work on the right track?

**Comment by sledz** [comment]
~~FYI: https://github.com/DFE/kms-windows/wiki/Further-development-of-the-KMS-Windows-port-for-H264-streaming-applications~~

**Comment by slimbyte** [comment]
@kirushyk @sledz @Stefan-Z-Camilleri @pschuessler @petergerten 
Hi guys,

Well, thanks to your snippets and notes I was able to compile KMS v. 6.7.1 for windows using MinGW-w64 via MSYS2 under Windows 7. After few weeks of work all seems to build just fine.

Now, I've set environment variable GST_DEBUG=4, to see some log data and started the kurento-hello-world sample. From the log file all seems to be ok, but no image on remote side, even if client console states "Loopback established". Also, when the client is connected, some errors are displayed on the server console:

![image](https://user-images.githubusercontent.com/3639033/40080605-1f37f26c-5894-11e8-847c-81c91516c570.png)

Log file attached
[2018-05-15_22-27-42.00000.pid1336.log](https://github.com/kirushyk/kms-windows/files/2006514/2018-05-15_22-27-42.00000.pid1336.log)

Any help would be appreciated!
Thanks.

Oh, if you want to have a look at my build you can download it from here:
https://drive.google.com/file/d/1iCX9q_TdEVMj9BMSC88XFgk7VfhF1WDK/view?usp=sharing

**Comment by slimbyte** [comment]
Please ignore the previous post. I just found my issue: I was missing libgstnice15.dll.
Added missing lib and now seems to work,
Thanks.

**Comment by kendziorra** [comment]
Hi Slimbyte,
could you publish your build to google drive as the broken one before?
Thanks!

**Comment by kirushyk** [comment]
Who is now manages development of Kurento Media Server? I think we should contact them and dedicate some time to merge Windows-related stuff into the their repository.

**Comment by slimbyte** [comment]
@kendziorra the gdrive repo was updated with working binaries; so, you can use same download link to download my latest build: 
~~https://drive.google.com/file/d/1iCX9q_TdEVMj9BMSC88XFgk7VfhF1WDK/view?usp=sharing~~

**Comment by sledz** [comment]
Hello @slimbyte, this is really great news. :+1:

As @kirushyk wrote, the next step would be to integrate the things related to the Windows build into the [upstream Kurento repositories](https://github.com/kurento) so as not to have just a one-shot release. Could you imagine doing this job? If so, how much effort do you expect? And when could this be completed?

In order to motivate you a little bit, we could well imagine offering a decent bonus (e.g. via [Bountysource](https://www.bountysource.com/) or another preferred way) for this task.

**Comment by slimbyte** [comment]
Hi @sledz,

I will be happy to do that job. 
The thing is, that I was hired by a third party to build latest Kurento sources for Windows. At this point they own my work, so, without their approval I can't do anything.

Thanks for understanding.

**Comment by kms4win** [comment]
@sledz I represent the third party slimbyte is talking about. Feel free to send me an email: kms@skylab-mobilesystems.com

**Comment by sledz** [comment]
Hello @slimbyte and @kms4win, we're not quite sure what "own my work" means in this case, probably it's a script or a build procedure.

Irrespective of this, we believe it makes sense for everyone involved if this build procedure is made available in general and actively maintained by the community in the future. This is the only way to ensure that future features or bug fixes will find their way into the KMS Windows port.

As I wrote above, we are willing to co-finance this task.

(@kms4win, regardless of this I'll contact you by email)

**Comment by kirushyk** [comment]
I think we should focus on making good contact with present Kurento team.

**Comment by sledz** [comment]
Ping @slimbyte, and @kms4win!

@kms4win, did you receive my mail?

**Comment by kms4win** [comment]
@sledz yes, I replied the same day - looks like you did not get my mail. Will send again.

**Comment by navidmafi** [comment]
hey everyone 
is there any plan for updating this project?
this is very old now , I even cannot find documentations for the API for this version
can a VPS with below specifications help? @kirushyk 
8vCore (main : AMD EPYC with IBPB)
16 GB RAM
3 GBPS download
1 GBPS upload
Hetzner Datacenter

it will expire soon , if it can help please tell me before this saturday to renewal
thanks 👌❤
![image](https://user-images.githubusercontent.com/46371349/109458605-5cfee080-7a72-11eb-986f-0abd4ed665c6.png)
